WebpageFunctions.py

```diff
@@ -0,0 +1,106 @@
+#######Imports#######
+import os
+import time
+"""
+Function that checks out the validation webpage branch from git
+Arguments:
+validation_path = The path to the WCSim validation directory
+travis_pull_request = state of the travis pull request ("true" or "false")
+"""
+def checkout_validation_webpage_branch(validation_path, travis_pull_request, travis_commit):
+    os.chdir(validation_path)
+
+    webpage_path = os.path.join(validation_path, "Webpage")
+
+    if not os.path.isdir(webpage_path):
+        os.system(f"git clone https://github.com/WCSim/Validation.git --single-branch --depth 1 -b gh-pages ./Webpage")
+        os.chdir("Webpage")
+
+        # Add a default user, otherwise git complains
+        os.system("git config user.name 'WCSim CI'")
+        os.system("git config user.email 'wcsim@wcsim.wcsim'")
+
+        os.chdir(f"{validation_path}")
+
+    travis_pull_request = str(travis_pull_request) if travis_pull_request is not None else "false"
+    print("Showing travis commit")
+    print(travis_commit)
+    print("Showing travis pull request")
+    print(travis_pull_request)
+
+"""
+Function that updates the webpage
+Arguments:
+validation_path = The path to the WCSim validation directory
+travis_commit
+GitHubToken
+"""
+def update_webpage(validation_path, travis_commit, GitHubToken):
+    os.chdir(os.path.join(validation_path, "Webpage"))
+
+    # Clean up old folders
+    folder = 0
+    with open(os.path.join(validation_path, "Webpage", "folderlist"), 'r') as folderlist_file:
+        for line in folderlist_file:
+            folder += 1
+            if folder >= 35:
+                os.system(f"git rm -r {line.strip()}")
+
+    # Setup the commit
+    print("Adding")
+    os.system("git add --all")
+    os.system(f"git commit -a -m 'CI update: new pages for {travis_commit}'")
+
+    # Setup a loop to prevent clashes when multiple jobs change the webpage at the same time
+    # Make it a for loop, so there isn't an infinite loop
+    # 100 attempts, 15 seconds between = 25 minutes of trying
+    # The CI is set up such that files will be touched by one job at once,
+    # so it's just a matter of keeping pulling until we happen to be at the front of the queue
+    for iattempt in range(101):
+        # Get the latest version of the webpage
+        os.system("git pull --rebase")
+
+        # Attempt to push
+        push_command = f"git push https://tdealtry:{GitHubToken}@github.com/WCSim/Validation.git gh-pages"

```

Comment:
Think it'd be good to have another config file that has (incomplete list)
- Validation
  - git path
  - git branch for code
  - git branch for webpage
  - folder to download webpage to
- Software being validated
  - Name
  - git path
  - git branch
- git commit
  - username
  - email

This'll be useful in the future for using the validation for different software, but also for now for testing (e.g. we can setup a new action to push to `gh-pages-test`)

Reply:
Indeed, this would be ideal. Also may elminate the need for some of the env variables. 

Can add in, will just need to tweak the `WebPageFuncs` class in `common.py` to reflect this!

Reply:
I've made a first go at this, but not sure what information to include, we should discuss :)